src/aliases/api/handlers/handlers.go

```diff
@@ -0,0 +1,297 @@
+// @Tags registry
+// @Param registry_name path string true "registry identifier"
+// @Success 204 {object} types.DeleteRegistryResponse "Registry data"
+// @Failure 400 {object} ErrorResponse "Invalid request format"
+// @Failure 404 {object} ErrorResponse "Registry not found"
+// @Failure 500 {object} ErrorResponse "Internal server error"
+// @Router /registries/{registry_name} [delete]
+func (h *AliasHandler) deleteRegistry(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	// should always exists in this subrouter
+	regName := vars["registry_name"]
+
+	err := validatePathVars(regName)
+	if err != nil {
+		infrahttp.WriteHTTPErrorResponse(w, errors.InvalidFormatError(err.Error()))
+		return
+	}
+
+	err = h.alias.DeleteRegistry(r.Context(), aliasent.RegistryName(regName))
+	if err != nil {
+		infrahttp.WriteHTTPErrorResponse(w, err)
+		return
+	}
+	w.Header().Set("X-Content-Type-Options", "nosniff")
+	w.WriteHeader(http.StatusNoContent)
+}
+
+// @Summary Creates an alias
+// @Description Create an alias of a key in a dedicated alias registry
+// @Tags alias
+// @Accept json
+// @Produce json
+// @Param registry_name path string true "registry identifier"
+// @Param alias_key path string true "alias identifier"
+// @Param request body types.AliasRequest true "Create Alias Request"
+// @Success 200 {object} types.AliasResponse "Alias data"
+// @Failure 400 {object} ErrorResponse "Invalid request format"
+// @Failure 500 {object} ErrorResponse "Internal server error"
+// @Router /registries/{registry_name}/aliases/{alias_key} [post]
+func (h *AliasHandler) createAlias(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	// should always exists in this subrouter
+	regName := vars["registry_name"]
+	key := vars["alias_key"]
+
+	err := validatePathVars(regName, key)
+	if err != nil {
+		infrahttp.WriteHTTPErrorResponse(w, errors.InvalidFormatError(err.Error()))
+		return
+	}
+
+	var aliasReq types.AliasRequest
+	err = jsonutils.UnmarshalBody(r.Body, &aliasReq)
+	if err != nil {
+		infrahttp.WriteHTTPErrorResponse(w, errors.InvalidFormatError(err.Error()))
+		return
+	}
+
+	eAlias := types.FormatAlias(types.RegistryName(regName), key, aliasReq.Value)
+	alias, err := h.alias.CreateAlias(r.Context(), eAlias.RegistryName, eAlias)
+	if err != nil {
+		infrahttp.WriteHTTPErrorResponse(w, err)
+		return
+	}
+
+	resp := types.AliasResponse{
+		Value: types.AliasValue(alias.Value),
+	}
+	err = infrahttp.WriteJSON(w, resp)
+	if err != nil {
+		infrahttp.WriteHTTPErrorResponse(w, err)
+		return
+	}
+}
+
+// @Summary Get an alias
+// @Description Get an alias of a key from a dedicated alias registry
+// @Tags alias
+// @Produce json
+// @Param registry_name path string true "registry identifier"
+// @Param alias_key path string true "alias identifier"
+// @Success 200 {object} types.AliasResponse "Alias data"
+// @Failure 400 {object} ErrorResponse "Invalid request format"
+// @Failure 404 {object} ErrorResponse "Alias not found"
```

Comment:
In case of registry not found, isn't it a 404 too?

Reply:
We actually don't have a registry, it's part of the same table, and it's the same composed key. So either/or, it will fail.
And since adding a key to an unexisting registry will create the "registry", then it will succeed.
It can either succeed if both ids are good, or it fails if any of them fails, we can't know which was wrong. But a good alias key in the wrong registry doesn't make it good anyway. So just a 404 saying about Alias should be good.

The real response says the resource is unavailable.
```
{"message":"ST100: resource not found","code":"ST100"}
```